lib/MusicBrainz/Server/Data/Recording.pm

```diff
@@ -112,6 +112,18 @@ sub find_by_artist
     $self->query_to_list_limited($query, \@where_args, $limit, $offset);
 }
 
+sub find_by_artist_credit
+{
+    my ($self, $artist_credit_id, $limit, $offset) = @_;
+
+    my $query = "SELECT DISTINCT " . $self->_columns . ",
```

Comment:
I don't think `DISTINCT` is needed? Since you don't have any joins and are selecting the primary key, each row will be unique.
```suggestion
    my $query = "SELECT " . $self->_columns . ",
```